documentation/src/main/asciidoc/topics/search.adoc

```diff
@@ -0,0 +1,1087 @@
+and after the grouping operation. All filter conditions defined before invoking the _groupBy_ method will be applied
+before the grouping operation is performed, directly to the cache entries (not to the final projection). These filter
+conditions may reference any fields of the queried entity type, and are meant to restrict the data set that is going to
+be the input for the grouping stage. All filter conditions defined after invoking the _groupBy_ method will be applied to
+the projection that results from the projection and grouping operation. These filter conditions can either reference any
+of the _groupBy_ fields or aggregated fields. Referencing aggregated fields that are not specified in the select clause
+is allowed; however, referencing non-aggregated and non-grouping fields is forbidden. Filtering in this phase will
+reduce the amount of groups based on their properties. Sorting may also be specified similar to usual queries. The
+ordering operation is performed after the grouping operation and can reference any of the _groupBy_ fields or aggregated
+fields.
+
+==== Using Full-text search
+
+===== Fuzzy Queries
+
+To execute a fuzzy query add `~` along with an integer, representing the distance from the term used, after the term.
+For instance
+
+[source,sql,tile="Fuzzy Queries in Ickle"]
+----
+FROM sample_bank_account.Transaction WHERE description : 'cofee'~2
+----
+
+===== Range Queries
+
+To execute a range query define the given boundaries within a pair of braces, as seen in the following example:
+
+[source,sql,tile="Range queries with Ickle"]
+----
+FROM sample_bank_account.Transaction WHERE amount : [20 to 50]
+----
+
+===== Phrase Queries
+
+A group of words may be searched by surrounding them in quotation marks, as seen in the following example:
+
+[source,sql,tile="Phrase queries with Ickle"]
+----
+FROM sample_bank_account.Transaction WHERE description : 'bus fare'
+----
+
+===== Proximity Queries
+
+To execute a proximity query, finding two terms within a specific distance, add a `~` along with the distance after the phrase.
+For instance, the following example will find the words canceling and fee provided they are not more than 3 words apart:
+
+[source,sql,tile="Proximity queries with Ickle"]
+----
+FROM sample_bank_account.Transaction WHERE description : 'canceling fee'~3
+----
+
+===== Wildcard Queries
+
+Both single-character and multi-character wildcard searches may be performed:
+
+* A single-character wildcard search may be used with the ? character.
+* A multi-character wildcard search may be used with the * character.
+
+To search for text or test the following single-character wildcard search would be used:
+
+[source,sql,tile="Single-character wildcard queries with Ickle"]
+----
+FROM sample_bank_account.Transaction where description : 'te?t'
+----
+
+To search for test, tests, or tester the following multi-character wildcard search would be useD:
```

Comment:
s/useD/used 

Consider changing from passive voice. "To search for "test", "tests", or "tester", use the following multi-character wildcard search:"